File: towerOfHanoi.java
public class towerOfHanoi {
    public static void main(String[] args) {
        towerofHanoi(3, 'A', 'B', 'C');
    }

    public static void towerofHanoi(int n, char source, char destination, char auxiliary) {
        if (n == 1) {
          System.out.println("Move disk 1 from rod " + source + " to rod " + destination);
          return;
        }
      
        towerofHanoi(n - 1, source, auxiliary, destination);
        System.out.println("Move disk " + n +" from rod " + source + " to rod " + destination);
        towerofHanoi(n - 1, auxiliary, destination, source);
    }
}

File: towerOfHanoi.py
def tower_of_hanoi(n, source, destination, auxiliary):
    if n == 1:
        print(f"Move disk 1 from rod {source} to rod {destination}")
        return
    tower_of_hanoi(n - 1, source, auxiliary, destination)
    print(f"Move disk {n} from rod {source} to rod {destination}")
    tower_of_hanoi(n - 1, auxiliary, destination, source)

# example usage
tower_of_hanoi(3, 'A', 'C', 'B')
